### DISCUSSION TITLE
slice a Blazor app into multiple sub apps

### DISCUSSION THREAD

**Discussion by Julien-Marpault** [dotnet/aspnetcore#24395] [General]
We will soon create a backoffice application, which is quite big.

So we are looking to the possibility to slice it into multiple Blazor apps.

Let's say the root application is named Discovery. his url is localhost:5001/home

and I want another application, let say a dashboard with url : localhost:5001/dashboard

I am looking for a way to do that either with standalone WASM apps and ASP Net core backed WASM apps

I tried to reproduce @javiercn exemple: https://github.com/javiercn/BlazorMultipleApps without success.

````csharp
app.MapWhen(
                ctx => ctx.Request.Path.ToString().Contains("dashboard"),
                dashboard =>
                {
                    dashboard.UseBlazorFrameworkFiles("/dashboard");
                    dashboard.UseStaticFiles();

                    dashboard.UseRouting();

                    dashboard.UseIdentityServer();
                    dashboard.UseAuthentication();
                    dashboard.UseAuthorization();

                    dashboard.UseEndpoints(endpoints =>
                    {
                        endpoints.MapRazorPages();
                        endpoints.MapControllers();
                        endpoints.MapFallbackToFile("/dashboard/{*path:nonfile}", "dashboard/index.html");
                    });
                }
            );

            app.MapWhen(
              ctx => ctx.Request.Path.ToString().Contains("home"),
              home =>
              {
                  home.UseBlazorFrameworkFiles("/home");
                  home.UseStaticFiles();

                  home.UseRouting();

                  home.UseIdentityServer();
                  home.UseAuthentication();
                  home.UseAuthorization();

                  home.UseEndpoints(endpoints =>
                  {
                      endpoints.MapRazorPages();
                      endpoints.MapControllers();
                      endpoints.MapFallbackToFile("/home/{*path:nonfile}", "home/index.html");
                  });
              }
          );
````
When doing that, both "/home" and "/dashboard" url reports 404 error and "/" url is hanging on loading message

Thanks

**Comment by xperiandri** [comment]
I want to achieve the same with the new .NET 8 mixed renderting model but I get `Conflicting assets with the same target path` from `GenerateStaticWebAssetsManifest` target.
@danroth27 is this approach achievable?

![image](https://github.com/dotnet/aspnetcore/assets/2365592/21141aeb-d0ac-45a3-a572-13b472e72145)
Violet -> server
Green -> common WASM libs
Red -> WASM apps to embed some interactivity on SSR pages

The idea was to load only requested piece of interactivity, not the whole single WASM project with all dependencies
